app/models/manageiq/providers/google/cloud_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,55 @@
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Runner <
+  ManageIQ::Providers::BaseManager::EventCatcher::Runner
+  include ManageIQ::Providers::Google::EventCatcherMixin
+
+  # Request the event monitor to stop running.
+  def stop_event_monitor
+    @event_monitor_handle.try(:stop)
+  ensure
+    reset_event_monitor_handle
+  end
+
+  # Start monitoring for events. This method blocks forever until
+  # #stop_event_monitor is called.
+  def monitor_events
+    _log.info "#{log_prefix} Monitoring for events"
+    event_monitor_handle.start
+    event_monitor_handle.each_batch do |events|
+      event_monitor_running
+      _log.debug "#{log_prefix} Received events #{events.collect { |e| parse_event_type(e) }}"
+      @queue.enq events
+      sleep_poll_normal
+    end
+  ensure
+    reset_event_monitor_handle
+  end
+
+  def process_event(event)
+    if filtered?(event)
+      _log.info(
+        "#{log_prefix} Skipping filtered Google event #{parse_event_type(event)} for #{parse_resource_id(event)}"
+      )
+    else
+      _log.info "#{log_prefix} Caught event #{parse_event_type(event)} for #{parse_resource_id(event)}"
+      EmsEvent.add_queue('add_google', @cfg[:ems_id], event)
+    end
+  end
+
+  private
+
+  def event_monitor_handle
+    unless @event_monitor_handle
```

Comment:
a common ruby idiom is using `||=` for something like this
```ruby
def event_monitor_handle
  @event_monitor_handle ||= ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream.new(@ems)
end
```

if thats too long
```ruby
def event_monitor_handle
  @event_monitor_handle ||= begin 
     ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream.new(@ems)
  end
end
```